cudax/include/cuda/experimental/__memory_resource/memory_pool_base.cuh

```diff
@@ -40,31 +40,169 @@
 namespace cuda::experimental
 {
 
-enum class __memory_location_type
+namespace __detail
 {
-  __device,
-  __host,
+template <::cudaMemPoolAttr _Attr, typename _Type, bool _Settable>
```

Comment:
For new APIs we settles on enumerations like `enum class _IsSettable : bool { _No, _Yes};`

That makes the APIs much easier to read at the call sites